### PR TITLE
Switch from fog to aws-sdk

### PR DESCRIPTION
**Pull request by ktheory**
Been bitten too many times by Fog dependency issues.

The code changes to use aws-sdk are pretty straightforward.

The tests aren’t great, and I couldn’t get stubbing to work correctly
for `Aws::S3::Object#exists?` so I skipped it.

I also refactored some insane testing setup.

Gonna test this manually before releasing.

cc @talaris, @jamtur01, and @cainlevy

### CODE REVIEW

File: lib/egads/s3_tarball.rb [hunk 1]
```diff
@@ -29,13 +29,13 @@ def local_tar_path
...
+        bucket.put_object(key: key, body: f)
       }
     end
 
-    # Load the file contents from S3
-    def contents
-      bucket.files.get(key).body
+    # Write the S3 object's contents to a local file
+    def download(file)
+      bucket.object(key).get(response_target: file)
```
Comment (ktheory) [inline lib/egads/s3_tarball.rb]: Easy refactor to stream the file download rather than load in all in memory.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -10,7 +10,8 @@
 rescue LoadError
 end
 
-Fog.mock!
+Fog.mock! # TODO: remove me
```
Comment (cainlevy) [inline spec/spec_helper.rb]: ?

File: lib/egads/s3_tarball.rb (hunk 1 above)
Comment (cainlevy) [inline lib/egads/s3_tarball.rb]: ooo

**Comment by cainlevy** [comment]
:dancers:

File: spec/spec_helper.rb (hunk 2 above)
Comment (ktheory) [inline spec/spec_helper.rb]: D'oh! Fixed: b6cdb12

**Comment by jamtur01** [comment]
LGTM

**Comment by jamtur01** [comment]
I might pull Fog out of ksr on the weekend too. But different yak a d different laser razor.

**Pull request merged**